### PR TITLE
feat: cli refactor

### LINKED ISSUE #680: feat: ci
fancy new ci

### PR DESCRIPTION
**Pull request by DOOduneye**
- [x] Still need to update the CONTRIBUTING.md and create documentation for the cli

### CODE REVIEW

File: cli/.golangci.yml
Comment (garrettladley) [inline cli/.golangci.yml:1]: why delete?

File: cli/LICENSE
Comment (garrettladley) [inline cli/LICENSE:1]: is this supposed to be empty?

File: cli/cmd/format.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	},
+}
+
+var formatBackendCmd = &cobra.Command{
+	Use:   "backend",
+	Short: "Backend formatting commands",
+	Aliases: []string{"be"},
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := helpers.Execute(exec.Command("go", "fmt", "./..."), helpers.BACKEND_DIR)
```
Comment (garrettladley) [inline cli/cmd/format.go:97]: why stray from `gofumpt -l -w .`?

File: cli/cmd/lint.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	},
+}
+
+var lintBackendCmd = &cobra.Command{
+	Use:   "backend",
+	Short: "Backend linting commands",
+	Aliases: []string{"be"},
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		// TODO: golangci-lint
```
Comment (garrettladley) [inline cli/cmd/lint.go:129]: marking this as a task. don't want a cli factor to be a regression.

`golangci-lint run --fix`

**Review by garrettladley — CHANGES_REQUESTED** [review]
some nits my 👑 also, should these frontend changes be in a diff PR? or are they as a result of formatting/linting with the updated CLI?

**Comment by garrettladley** [comment]
also @DOOduneye, is the tab autocomplete working or should we open an issue to track that feature?

**Comment by DOOduneye** [comment]
> also @DOOduneye, is the tab autocomplete working or should we open an issue to track that feature?

im going to make an issue, i thought it was as simple as `sac completion zsh > /tmp/completion` but that doesn't work

**Comment by DOOduneye** [comment]
> some nits my 👑 also, should these frontend changes be in a diff PR? or are they as a result of formatting/linting with the updated CLI?

frontend changes come from running and testing the cli

**Pull request merged**